WordPress/Classes/Services/AccountService.m

```diff
@@ -35,22 +35,20 @@ @implementation AccountService
  */
 - (WPAccount *)defaultWordPressComAccount
 {
-    NSString *uuid = [[NSUserDefaults standardUserDefaults] stringForKey:DefaultDotcomAccountUUIDDefaultsKey];
-    if (uuid.length == 0) {
+    NSURL *myObjectURL = [NSURL URLWithString:[[NSUserDefaults standardUserDefaults] objectForKey:DefaultDotcomAccountIDDefaultsKey]];
+    NSManagedObjectID *myObjectID = [self.managedObjectContext.persistentStoreCoordinator managedObjectIDForURIRepresentation:myObjectURL];
+
+    if (!myObjectID) {
         return nil;
     }
 
-    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Account"];
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"uuid == %@", uuid];
-    fetchRequest.predicate = predicate;
-    
     NSError *error = nil;
-    NSArray *fetchedObjects = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    WPAccount *foundAccount = [self.managedObjectContext existingObjectWithID:myObjectID error:&error];
```

Comment:
Ok thanks.
I'm thinking about going back to the previous plan to use userId, and now that self-account won't setDefaultAccount nor use WPAccount it shouldn't be an userId collision. But I have a questions:

Do you think we might have trouble changing the moment were we set the `defaultAccount` which currently is as soon as we get the authenticationAccount response (for wp.com accounts) and before calling `updateUserDetailsForAccount`, to when we receive the response for `updateUserDetailsForAccount`? 
So at that point we will have the userId, and I think we won't have a migration issue using it instead of UUID

Thoughts @koke @jleandroperez